From 755c7c9c4f39b690892d2b38a2fd44f4dd602d4a Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 3 Jul 2019 20:37:14 +0100
Subject: [PATCH 683/773] overlays: Correct gpio-fan gpio flags for 4.19

The gpio-fan overlay was submitted for the 4.14 kernel where the second
value in the Device Tree gpios declaration was ignored (thanks to an
old-style driver), allowing the fan-control output to be active-high
even though the declaration appears to request it be active-low.
The gpio-fan driver in 4.19 uses GPIO descriptors and honours the
active-low flag that the overlay (accidentally?) supplies.

Change/correct the flags field to mark the GPIO as active-high.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/gpio-fan-overlay.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/arm/boot/dts/overlays/gpio-fan-overlay.dts
+++ b/arch/arm/boot/dts/overlays/gpio-fan-overlay.dts
@@ -45,7 +45,7 @@
 		__overlay__ {
 			fan0: gpio-fan@0 {
 				compatible = "gpio-fan";
-				gpios = <&gpio 12 1>;
+				gpios = <&gpio 12 0>;
 				gpio-fan,speed-map = <0    0>,
 									 <5000 1>;
 				#cooling-cells = <2>;
